src/Principal.mo

```diff
@@ -1,58 +1,241 @@
-/// IC principals (user and canister smart contract IDs)
+/// Module for interacting with Principals (users and canisters).
+///
+/// Principals are used to identify entities that can interact with the Internet
+/// Computer. These entities are either users or canisters.
+///
+/// Example textual representation of Prinicpals:
+///
+/// `un4fu-tqaaa-aaaab-qadjq-cai`
+///
+/// In Motoko, there is a primitive Principal type called `Principal`. As an example
+/// of where you might see Prinicpals, you can access the Principal of the
+/// caller of your shared function.
+///
+/// ```motoko no-repl
+/// shared(msg) func foo() {
+///   let caller : Principal = msg.caller;
+/// };
+/// ```
+///
+/// Then, you can use this module to work with the `Principal`.
+///
+/// Import from the base library to use this module.
+/// ```motoko name=import
+/// import Principal "mo:base/Principal";
+/// ```
 
 import Prim "mo:⛔";
 import Blob "Blob";
 import Hash "Hash";
 module {
 
-  /// Internet Computer principal identifiers.
-  /// Convert to `Blob` for access to bytes.
   public type Principal = Prim.Types.Principal;
 
-  /// Conversion.
+  /// Get the `Principal` identifier of an actor.
+  ///
+  /// Example:
+  /// ```motoko include=import no-repl
+  /// actor MyCanister {
+  ///   let principal = Principal.fromActor(MyCanister);
```

Comment:
```suggestion
  ///   public func getPrincipal() : async Principal { 
  ///        Principal.fromActor(MyCanister)
  ///   }
```
Unfortunately, this should produce an error at the moment because MyCanister is consider undefined in the constructor.
The example above should type check.

Reply:
(we have plans to fix this https://github.com/dfinity/motoko/issues/3718 since it is annoying)